coalib/results/TextPosition.py

```diff
@@ -33,6 +33,19 @@ def column(self):
         return self._column
 
     @classmethod
+    def calc_position(self, text, line, column):
+        """
+        Calculates position in a string from line number and column.
+        Uses \\n  as the newline character. Position starts from 0
+        """
+        lines = text.splitlines(keepends=True)
+        position = 0
+        for line_nr in range(0, line-1):
+            position += len(lines[line_nr])
```

Comment:
This is basically a cumsum. I'd recomment not doing a for loop (they're slow).
You can just a generator with `sum()`, it'd be faster.

PS: For more extreme optimization http://stackoverflow.com/questions/9258602/elegant-pythonic-cumsum